framework/src/node/consensus/certificate_generation/commit_pool.ts

```diff
@@ -62,15 +64,103 @@ export class CommitPool {
 
 	// eslint-disable-next-line @typescript-eslint/no-empty-function
 	public async job(): Promise<void> {}
-	// eslint-disable-next-line @typescript-eslint/no-empty-function
-	public addCommit(_commit: SingleCommit, _height: number): void {}
-	// eslint-disable-next-line @typescript-eslint/no-empty-function
-	public validateCommit(_commit: SingleCommit): boolean {
+
+	public addCommit(commit: SingleCommit, height: number): void {
+		const currentCommits = this._nonGossipedCommits.get(height) ?? [];
+		const doesCommitExist = currentCommits.some(aCommit =>
+			aCommit.certificateSignature.equals(commit.certificateSignature),
+		);
+		if (!doesCommitExist) {
+			this._nonGossipedCommits.set(height, [...currentCommits, commit]);
+		}
+	}
+
+	public async validateCommit(apiContext: APIContext, commit: SingleCommit): Promise<boolean> {
+		// Validation step 4
+		const blockHeaderAtCommitHeight = await this._chain.dataAccess.getBlockHeaderByHeight(
+			commit.height,
+		);
+		if (!blockHeaderAtCommitHeight.id.equals(commit.blockID)) {
+			return false;
+		}
+
+		// Validation step 1
+		const doesCommitExistsInNonGossipedCommits = !!this._nonGossipedCommits
+			.get(commit.height)
+			?.some(
```

Comment:
shouldn't we use `.find` as we always have unique commits for a height (we always discard duplicates while adding)

Reply:
I wanted something to return boolean instead of the item itself. Since we cast the result to boolean by double negation, it actually doesn't matter. If you really think I should change it to `find`, I can.